### DISCUSSION TITLE
Should setting a user 0 volume level send mute indicator to their client

### DISCUSSION THREAD

**Discussion by rdica** [jamulussoftware/jamulus#2727] [General]
As I understand it, on current versions of Jamulus client, if you mute a user, that user will see a spkr icon above the muters channel in their client, indicating they're being muted.
If one were to set a users volume to 0, that too will show the spkr icon indicating being muted.
Initially I thought using the mute button on a user was the only way that user would know they were being muted.
I discovered during testing of New Client Level 0% that this too shows the other users they're being muted.

I feel this sends the wrong message to others as they may think they're being muted due to some misbehaviour or other issues.
"_Why is that person muting me? I've never interacted with them before._"
There's a punittive impression given in my opinion.
My feeling is that another user should see a muted indicator only if someone actually used the mute button, and not just zeroing the volume level. Those that set New Client Level to 0% wouldn't be indicating to every new client encountered that they are being muted.

Is this the intended behaviour?

Do we want a differentiation between muting and zeroing volumes as far as what anothers client sees?

Thoughts?

**Comment by chrisrimple** [comment]
What is the purpose of the Mute indicator?  To (1) help a user understand that they've been (likely accidentally) muted when they can't be heard by another performer or to (2) help a user understand that they've been muted so they can be upset at the other performer?  I'm pretty sure that it's (1), so showing the Mute icon for Volume = 0% seems appropriate.

And really, if a performer has set Volume = 0%, they intend to Mute that user, so why not show it as such?

**Reply by ghost** [reply]
> What is the purpose of the Mute indicator? To (1) help a user understand that they've been (likely accidentally) muted when they can't be heard by another performer or to (2) help a user understand that they've been muted so they can be upset at the other performer? I'm pretty sure that it's (1), so showing the Mute icon for Volume = 0% seems appropriate.
> 
> And really, if a performer has set Volume = 0%, they intend to Mute that user, so why not show it as such?

I think the Mute Indicator should be removed as neither (1) or (2) is useful. (1) is useful only when everyone has accidentally muted someone.

**Comment by pljones** [comment]
The current implementation of Jamulus has poor separation between the UI and the implementation.  Some things are "done in the UI" - where the client itself doesn't really know what's being done behind its back.

The mixer, remember, is on the server.  The client only has the ability to tell the server the level for each channel in the mix.

The server has no concept of solo or mute.  These are client-side concepts.

Worse than that -- they appear to be Client UI concepts.  The client itself doesn't know about them.  It's just told what to do by the UI and, in turn, tells the server what to do.

So if you click "Solo", everyone you didn't solo is told you muted them.

And if you click "Mute", that person is told you muted them.

But because these are, essentially, simply representations of "your level in someone's mix is zero" (all the server knows and all it can tell you), if you drag the level down to zero for someone, they'll also get told they're muted.

If that's how it's working in beta1, that's considered correct.

**Reply by rdica** [reply]
> So if you click "Solo", everyone you didn't solo is told you muted them.

Ahhhhh I wondered about this, thanks for the clarification.
Yes 3.9.0beta1 does indeed continue to behave as older versions in this respect.
I was thinking there was something client side that "sent" a "muted" notification and that it could possibly be altered depending on how the volume level was handled, muted vs zero'd. Working under this assumption I was intending to initiate a discussion surrounding this.

Ok since this isn't something easily handled, I don't see a point to belabour this any further.

Thanks @pljones and @chrisrimple for your input.
